src/services/composite-keys-manager.js

```diff
@@ -1,54 +1,92 @@
 import _ from 'lodash';
+import { Schemas } from 'forest-express';
 import Operators from '../utils/operators';
 
-function CompositeKeysManager(model, schema, record) {
-  const GLUE = '|';
+/**
+ * This helper class allows abstracting away the complexity
+ * of using collection which have composite primary keys.
+ */
+class CompositeKeysManager {
+  constructor(model, glue = '|') {
+    this._glue = glue;
+    this._model = model;
+    this._schema = Schemas.schemas[model.name];
+    this._Sequelize = this._model.sequelize.constructor;
+  }
 
-  this.getPrimaryKeyValues = function getPrimaryKeyValues(recordId) {
-    const primaryKeyValues = recordId.split(GLUE);
+  /** Build sequelize where condition from a list of packed recordIds */
+  getRecordsConditions(recordIds) {
+    const Ops = Operators.getInstance({ Sequelize: this._Sequelize });
 
-    // NOTICE: Prevent liana to crash when a composite primary keys is null,
-    //         this behaviour should be avoid instead of fixed.
-    primaryKeyValues.forEach((key, index) => {
-      if (key === 'null') { primaryKeyValues[index] = null; }
-    });
+    // Empty array => return a query that match no records instead of crashing.
+    if (recordIds.length === 0) {
+      throw new Error('At least one recordId must be provided');
+    }
 
-    return primaryKeyValues;
-  };
+    // If the ids are not composite, use a IN.
+    if (!this._schema.isCompositePrimary) {
+      return { [this._schema.idField]: recordIds };
+    }
 
-  this.getRecordsConditions = function getRecordsConditions(recordsIds, options) {
-    const { OR } = Operators.getInstance(options);
-    return { [OR]: recordsIds.map((recordId) => this.getRecordConditions(recordId)) };
-  };
+    return recordIds.length === 1
+      ? this._getRecordConditions(recordIds[0])
+      : { [Ops.OR]: recordIds.map((recordId) => this._getRecordConditions(recordId)) };
+  }
 
-  this.getRecordConditions = function getRecordConditions(recordId) {
-    const where = {};
-    const primaryKeyValues = this.getPrimaryKeyValues(recordId);
-    if (primaryKeyValues.length === _.keys(model.primaryKeys).length) {
-      _.keys(model.primaryKeys).forEach((primaryKey, index) => {
-        where[primaryKey] = primaryKeyValues[index];
+  /* Annotate records with their packed primary key */
+  annotateRecords(records) {
+    if (this._schema.isCompositePrimary) {
+      records.forEach((record) => {
+        record.forestCompositePrimary = this._createCompositePrimary(record);
       });
     }
+  }
+
+  /** Build sequelize where condition from a single packed recordId */
+  _getRecordConditions(recordId) {
+    const primaryKeyValues = this._getPrimaryKeyValues(recordId);
+
+    // Invalid id => return no records.
+    if (primaryKeyValues.length !== _.keys(this._model.primaryKeys).length) {
+      return this._Sequelize.literal('(0=1)');
+    }
```

Comment:
This function is supposed to return a where condition that matches a given composite id.

The previous implementation returned an empty clause when an invalid id was passed (which means "match all records").
This bugfix returns a clause that will never be true in SQL so that no rows match.

I made this so that when we call "DELETE /forest/mycollection?ids[]=1|1|1|1|1" no record get deleted (currently, all records get deleted)

I could have used any string that evaluates to false in SQL and which is supported everywhere (mssql, oracle, ...)